### ISSUE TITLE
urgency: update front end utiltiities

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#3391]
With the rename of administrative priority to "urgency", we need a few changes to our front end utilities:

`flux job submit --priority=N`
Rename option to `--urgency`?

`flux mini {run|submit|batch|alloc} --priority=N` 
Rename option to `--urgency`?

`flux job priority ID PRIORITY`
Rename su-bcommand to `urgency`?

Idea: @grondo mentioned that he would like to see a way to raise or lower the urgency e.g. `flux job urgency ID +1`.

Idea: Change the job sub-command to  `update` and then allow debug flags to be updated as well, e.g.
```
$ flux job update --urgency=+1 ID
$ flux job update --flags=+waitable ID
```

Thoughts?

**Comment by chu11** [comment]
in terms of the renaming, i'm taking care of many of these via #3380.

**Comment by chu11** [comment]
as for the update subcommand, i like the idea.  inevitably there will probably be many updates that can be done to jobs, and we probably do not want to pollute `flux job` with more and more subcommands.  But should it be sub-sub-commands instead of flags?  i.e.

```
$ flux job update urgency +1 ID
$ flux job update flags +waitable ID
```

b/c underneath the covers, it may be different RPCs to different modules of flux?

**Comment by grondo** [comment]
One drawback to using sub-sub-commands is that you can't update multiple things in one call.

**Comment by grondo** [comment]
Though the command could be more flexible by using `key=val` syntax, allowing multiple per line:

```console
$ flux job update ID urgency=+1 flags=+waitable duration=+10m
```

Probably all updates made in this way would go to a job update service in the job-manager, which would handle updating any other services if necessary (and would post an update event to eventlog)

**Comment by chu11** [comment]
per discussion in #3428, generalize this `flux job update` to include a modify RPC.  per comments from @garlick 


```C
flux_future_t *flux_job_modify_pack  (flux_t *h, flux_jobid_t id, const char *name, const char *fmt, ...);
int flux_job_modify_get_unpack (flux_future_t *f, const char *fmt, ...);
```

```C
f = flux_job_modify_pack (h, id, "urgency", "{s:b s:i}", "relative", 1, "value", -1);
flux_job_modify_get_unpack (f, "{s:s}", "value", &last_value);
```

**Comment by garlick** [comment]
Could be `flux_job_update_pack()` if we prefer update to modify.  No strong opinion here.  Actually I like update 0.1% better.

**Comment by chu11** [comment]
FWIW, initially added "+/-N" support to `flux job urgency` for #3427 in this branch.  Could yoink some of the code / tests in future.

https://github.com/chu11/flux-core/tree/issue2895_issue3324_job_hold_expedite_try3

I preferred using the parameter "operation" instead of "relative".

```
enum job_urgency_operation {
    FLUX_URGENCY_OPERATION_SET      = 1,
    FLUX_URGENCY_OPERATION_ADD      = 2,
    FLUX_URGENCY_OPERATION_SUBTRACT = 4,
};
```

**Comment by grondo** [comment]
I actually feel like relative is simpler. Or just use "add" and allow negative values. It doesn't really feel like the enum `job_value_operation` would be adding much to the header file.

If we wanted to me more flexible, we could put the name of values being modified as keys in the payload, allowing multiple values to be updated in one RPC.

**Comment by chu11** [comment]
> If we wanted to me more flexible, we could put the name of values being modified as keys in the payload, allowing multiple values to be updated in one RPC.

Yeah, i actually imagined an array of name/relative/value modification requests if we suddenly have many fields to modify.  But in the mean time, I think the proposal from @garlick is fine b/c we just don't have that many fields.
